CADETProcess/comparison/difference.py

```diff
@@ -530,7 +532,7 @@ def __init__(
             )
             self.reference_der_sliced = slice_solution(
                 self.reference_der,
-                None,
+                self.components,
```

Comment:
To be honest, I'm not sure this is a good idea. This way, we "force" our Reference to have the same component system as our solution. While more consistent, this also comes with some downsides:
- it is a bit more cumbersome for the user to set up
- it could be a limitation when we are comparing to some sum-signal (e.g. total concentration)

so maybe we should reverse this change.

There were some issues during testing but maybe they came from somewhere else; I'll have to re-check...